code/implementation/docker/machine/src/main/java/io/cattle/platform/docker/machine/launch/CatalogLauncher.java

```diff
@@ -40,12 +111,27 @@ protected String binaryPath() {
     @Override
     protected void prepareProcess(ProcessBuilder pb) {
         List<String> args = pb.command();
-        args.add("-catalogUrl");
-        args.add(CATALOG_URL.get());
+        args.add("-configFile");
+        prepareConfigFile();
+        args.add("repo.json");
         args.add("-refreshInterval");
         args.add(CATALOG_REFRESH_INTERVAL.get());
     }
 
+    protected void prepareConfigFile() {
+        String catUrl = CATALOG_URL.get();
+
+        File configFile = new File("repo.json");
+        try {
+            ConfigFileFields configCatalogEntries = jsonMapper.readValue(catUrl, ConfigFileFields.class);
+            FileOutputStream fos = new FileOutputStream(configFile.getAbsoluteFile());
+            jsonMapper.writeValue(fos, configCatalogEntries);
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
```

Comment:
This error should be properly handled.  Just add throws to the method so that it bubbles up.  Let the framework handle it.